### PR TITLE
 Add an experimental feature gate for global registration

### PR DESCRIPTION
**Pull request by jdonszelmann**
See #125119 for the tracking issue.

### CODE REVIEW

**Comment by jdonszelmann** [comment]
(I think it's appropriate to tag @scottmcm in this, I'm sorry if you didn't want to Scott)

File: compiler/rustc_feature/src/unstable.rs [hunk 1]
```diff
@@ -487,6 +487,8 @@ declare_features! (
     (incomplete, generic_const_exprs, "1.56.0", Some(76560)),
     /// Allows generic parameters and where-clauses on free & associated const items.
     (incomplete, generic_const_items, "1.73.0", Some(113521)),
+    /// Allows registring static items globally, possibly across crates, to iterate over at runtime.
```
Comment (pnkfelix) [inline compiler/rustc_feature/src/unstable.rs:490]: nit: you have misspelled "registering" here.

**Comment by pnkfelix** [comment]
this can be r=me once the nit I noted is addressed. :)

File: compiler/rustc_feature/src/unstable.rs (hunk 1 above)
Comment (pnkfelix) [reply compiler/rustc_feature/src/unstable.rs:490]:
```suggestion
    /// Allows registering static items globally, possibly across crates, to iterate over at runtime.
```

**Comment by pnkfelix** [comment]
(actually I decided roundtripping through the author here is silly. I just did it myself.)

**Comment by jdonszelmann** [comment]
Thanks! Also for the fix haha

**Pull request merged**